crates/bevy_input_focus/src/directional_navigation.rs

```diff
@@ -184,6 +197,168 @@ impl DirectionalNavigationMap {
         }
     }
 
+    /// Adds edges to a grid of entities, connecting each entity to its neighbors in each of the eight directions.
+    ///
+    /// The grid is provided as a hashmap of [`IVec2`] to entities,
+    /// where the `x` represents columns from the top and `y` represents rows from the left.
+    /// Missing entities will be ignored, so sparse grids are supported.
+    /// If an element is missing, their neighbors will be connected to the nearest existing entity in the given direction.
+    ///
+    /// Diagonals travel in a purely 1:1 step fashion, so an entity at (0, 0) will connect to an entity at (2, 2)
+    /// in the [`SouthEast`](CompassOctant::SouthEast) direction, but would never see an entity at (1, 2).
+    /// That said, you can add the same entity at multiple locations in the grid!
+    /// This works well to model larger entities,
+    /// even if the entities aren't truly grid-aligned.
+    ///
+    /// If `should_loop` is set to `true`, the top and bottom rows will be connected, as will the left and right columns.
+    // The keys are specified as i32 for compatibility with taffy's grid system,
+    // which uses u16 values. We need signed integers here, but we should be able to cast them losslessly.
+    // PERF: This is a very flexible / "clever" implementation, but it won't be the fastest for simple cases.
+    // If there's user demand for it, we can add a more optimized / less flexible version that requires non-sparse rectangular grids.
+    pub fn add_grid(&mut self, entity_grid: NavGrid, should_loop: bool) {
```

Comment:
This promises support for placing entities at multiple locations, however it's not well defined what the result is.
Consider the following grid:
```
X A
Y A
```
Now... what is the left neighbour of A? It seems to be completely implementation dependent.

Reply:
Yeah, we should probably choose, document and test a rule for this.